File: src/test/java/com/raychenon/hackerrank/AVeryBigSumTest.java
package com.raychenon.hackerrank;

import org.junit.Test;

import java.util.List;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 13/1/19
 */
public class AVeryBigSumTest {
    
    @Test
    public void assertSum() {
        long[] array1 = {1, 2, 3, 4, 5};
        assertEquals(AVeryBigSum.aVeryBigSum(array1), 15);

        long[] array2 = {1000000001, 1000000002, 1000000003, 1000000004, 1000000005};
        assertEquals(AVeryBigSum.aVeryBigSum(array2), 5000000015L);
    }

    @Test
    public void assertSumWithEmptyArray() {
        long[] emptyArray = {};
        assertEquals(AVeryBigSum.aVeryBigSum(emptyArray), 0);
    }

}

File: src/test/kotlin/com/raychenon/kotlin/hackerrank/AVeryBigSumTest.kt
package com.raychenon.kotlin.hackerrank

import org.junit.Assert
import org.junit.Test

/**
 * User: raychenon
 * Date: 14/1/19
 */
class AVeryBigSumTest {

    @Test
    fun evaluateSums(){
        Assert.assertEquals(AVeryBigSumKotlin.aVeryBigSum(longArrayOf(1, 2, 3, 4, 5).toTypedArray()), 15)
        Assert.assertEquals(AVeryBigSumKotlin.aVeryBigSum(longArrayOf(6, 7, 8, 9,10).toTypedArray()), 40)
        Assert.assertEquals(AVeryBigSumKotlin.aVeryBigSum(longArrayOf(1000000001, 1000000002, 1000000003, 1000000004, 1000000005).toTypedArray()), 5000000015)
    }

    @Test
    fun evaluateCountSums(){

        Assert.assertEquals(AVeryBigSumKotlin.aVeryBigSum(LongArray(100){i -> 1}.toTypedArray()), 100)

        Assert.assertEquals(AVeryBigSumKotlin.aVeryBigSum(LongArray(100){i -> 2}.toTypedArray()), 200)
    }
}